### ISSUE TITLE
Provider version incompatible with darwin_arm64

### ISSUE THREAD

**Issue by nottooshort** [cyralinc/terraform-provider-cyral#160]
MacOSX Version: 12.0.1
MacBook Pro (14-inch, 2021)

```
Terraform v1.1.2
on darwin_arm64
+ provider registry.terraform.io/cyralinc/cyral v2.4.3
+ provider registry.terraform.io/okta/okta v3.20.2

Your version of Terraform is out of date! The latest version
```

$ terraform init
```

│ Error: Incompatible provider version
│
│ Provider registry.terraform.io/cyralinc/cyral v2.4.3 does not have a package available for your current platform, darwin_arm64.
│
│ Provider releases are separate from Terraform CLI releases, so not all providers are available for all platforms. Other versions of this provider may have
│ different platforms supported.
```

I don't see anything in the changelog that indicates that the latest cryal provider supports darwin_arm64

**Comment by greenkiwi** [comment]
As a workaround, we are just locking the version to AMD64.  This is the script we are using to update our repos: 
```
#!/usr/bin/env bash

set -e

REPO_DIR="$(dirname "$( cd "$( dirname "${BASH_SOURCE[0]}" )" >/dev/null 2>&1 && pwd )")"

# find all directories containing .terraform.lock.hcl files
LOCK_FILES=$(find "${REPO_DIR}" -name '.terraform.lock.hcl')

# for each LOCK_FILE, extract the provider name and the version
LOCK_FILE_DIRS=$(for LOCK_FILE in ${LOCK_FILES}; do
  dirname "${LOCK_FILE}"
done)

# for each LOCK_FILE_DIR, extract the provider name and the version
for LOCK_FILE_DIR in ${LOCK_FILE_DIRS}; do
  echo "Updating ${LOCK_FILE_DIR}"
  terraform -chdir="$LOCK_FILE_DIR" init --upgrade
  terraform -chdir="$LOCK_FILE_DIR" providers lock -platform=darwin_amd64 -platform=linux_amd64
done

```

**Comment by wcmjunior** [comment]
Fixed by #181 . Validated in the following machine: 
![image](https://user-images.githubusercontent.com/796900/161164537-ae502c6e-679b-4b82-b325-a49c78bc8383.png)

**Comment by wcmjunior** [comment]
Fix available since `v2.5.1`.
